src/Cache/Engine/FileEngine.php

```diff
@@ -388,7 +388,7 @@ protected function _setKey($key, $createKey = false)
         if (!$createKey && !$path->isFile()) {
             return false;
         }
-        if (empty($this->_File) || $this->_File->getBasename() !== $key) {
+        if (null === $this->_File || $this->_File->getBasename() !== $key) {
```

Comment:
my 2 cents on Yoda:

It creates a longer thought process. You bascially read:
1. Entity null
2. has to equal something
1.+2. entity null has to equal something
3. $this->_File
4. Entity Null has to Equal $this->_File

It is like double negation. You go one extra circle in your through process. You have to read from right to left to circumvent that.

When it is $foo === generic_entity/statement (such as null, false, true) it is much easier than the other way around.

Also for comparison I feel many peeps still "assign" things under the condition of that it can be assigned. "Can Entity Null be assigned to the Equality of some other Entity"? That's a hard way to look at it ... "Can Entity X be assigned to the Equality of Entity Null" is a lot easier for my mind.

Anyway, I don't think this PR got rejected because of YODA style as that can easily be fixed.